### PR TITLE
feat(spdk): spdk bindings improvements

### PR DESCRIPTION
**Pull request by dsavitskiy**
* NVME status code refactored to refer directly to SPDK enums
* NvmfController simple wrapper added
* NvmfSubsystemEvent support added (requires corresponding SPDK changes)
* Bdev and Bdev I/O debug fmt improvements

This is needed to improve SPDK-RS in general, and to implement support for setting CRD selectively.

### CODE REVIEW

File: src/nvmf.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            ),
+            SPDK_NVMF_SUBSYSTEM_EVENT_HOST_DISCONNECT => Self::HostDisconnect(
+                NvmfController::from(ctx as *mut spdk_nvmf_ctrlr),
+            ),
+            SPDK_NVMF_SUBSYSTEM_EVENT_HOST_KEEP_ALIVE_TIMEOUT => {
+                Self::HostKeepAliveTimeout(NvmfController::from(
+                    ctx as *mut spdk_nvmf_ctrlr,
+                ))
+            }
+            _ => panic!("Unknown NVMF subsystem event: {:?}", event),
```
Comment (tiagolobocastro) [inline src/nvmf.rs]: Should this be a try_from kinda function, otherwise if spdk adds a new event we might fall into a panic here

**Review by tiagolobocastro — APPROVED** [review]

File: src/bdev.rs [hunk 2]
```diff
@@ -377,6 +377,36 @@ where
...
+                .field("aliases", &self.aliases())
+                .field("num_blocks", &self.num_blocks())
+                .field("block_len", &self.block_len())
+                .field("claimed_by", &self.first_claim_module_name())
+                .field("ptr", &unsafe { self.unsafe_inner_ptr() })
+                .finish()
+        } else {
+            write!(
+                f,
+                "{m}::{name} {sz}x{blk}",
```
Comment (dsharma-dc) [inline src/bdev.rs]: nitpick: Perhaps say "{m}::{name} {sz}x{blk} (bytes)"

File: src/bdev_io.rs [hunk 3]
```diff
@@ -133,15 +133,19 @@ where
...
-    /// -- offset where we do the IO on the device
+    /// Offset of the first block where we do the I/O on the device.
     #[inline]
     pub fn offset(&self) -> u64 {
         unsafe { self.as_ref().u.bdev.offset_blocks }
     }
 
-    /// TODO
-    /// -- num of blocks this IO will read/write/unmap
+    /// Offset of the last block where we do the I/O on the device.
```
Comment (dsharma-dc) [inline src/bdev_io.rs]: I guess this is going to return the offset of the block which is right after the last block of current I/O. Is that what's intended here?

Comment (dsavitskiy) [reply src/bdev_io.rs]: It's actually a pointless method I added. I'll just remove it.

File: src/nvmf.rs (hunk 1 above)
Comment (dsavitskiy) [reply src/nvmf.rs]: I changed to warn! and returning an Unknown event type.

**Pull request merged**